doc: piping from async generators using pipeline() (#33992)
nodejs/node: doc/api/stream.md

#### Piping to writable streams from async iterators

> In the scenario of writing to a writable stream from an async iterator, the
> correct handling of backpressure and errors must be ensured.

Reviewer: A case can be made here for the switch to passive voice but maybe consider keeping it as active voice?:

```suggestion
When writing to a writable stream from an async iterator, ensure correct
handling of backpressure and errors.
```

Author: Sure, that works for me! I have changed this.

> [`stream.pipeline()`][] conveniently abstracts away the handling of

Reviewer: Another optional suggestion, but I try to avoid adverbs if they're not necessary.

```suggestion
[`stream.pipeline()`][] abstracts away the handling of
```

Author: Done. Thanks for the tip!

backpressure and backpressure-related errors:

```js
const { pipeline } = require('stream');
const util = require('util');
const fs = require('fs');

const writable = fs.createWriteStream('./file');

// Callback Pattern
pipeline(iterator, writable, (err, value) => {
  if (err) {
    console.error(err);
  } else {
    console.log(value, 'value returned');
  }
});

// Promise Pattern
const pipelinePromise = util.promisify(pipeline);
pipelinePromise(iterator, writable)
  .then((value) => {
    console.log(value, 'value returned');
  })
  .catch(console.error);
```

Although this is unnecessary if using [`stream.pipeline()`][], it is possible
to manually handle backpressure and backpressure-related errors:

> 

Reviewer 2: I think we should remove the part below. Makes things complicated.

Reviewer 3: I agree. I would not recommend folks to use it at all.

Author: Sounds good. I have removed that section completely.
